external/fv3fit/fv3fit/_shared/packer.py

```diff
@@ -72,7 +80,7 @@ def __init__(self, sample_dim_name, pack_names: Iterable[Hashable]):
         self._pack_names: List[str] = list(str(s) for s in pack_names)
         self._n_features: Dict[str, int] = {}
         self._sample_dim_name = sample_dim_name
-        self._dims: Dict[str, Sequence[str]] = {}
+        self._feature_index: pd.MultiIndex = pd.MultiIndex(levels=[[]], codes=[[]])
```

Comment:
Is this default value useful? If not, can you please make this Optional and add logic to raise an exception if it isn't initialized? Otherwise this could be used and cause silent errors. If you expect this to cause loud errors when used, that can be OK as well.

If you want to avoid putting if checks, you can make a `feature_index` property method which raises an exception when accessed if this hasn't been initialized.

Reply:
This isn't really a new issue... the same could be said for the other attrs like `self._dims` and `self._n_features` which only really get 'initialized' on the first `to_array` call. Anyway, I'll raise an exception if someone tries to unpack before packing, but I would like to refactor the API of ArrayPacker as we discussed over slack to remove the possibility of this issue anyway.